llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/MaterializeEncoding.cpp

```diff
@@ -400,21 +400,20 @@ MaterializeEncodingTypeConverter::MaterializeEncodingTypeConverter(
 MaterializeEncodingConversionTarget::MaterializeEncodingConversionTarget(
     MLIRContext &context)
     : ConversionTarget(context) {
-  auto typeHasEncoding = [=](Type t) -> bool {
-    auto tensorType = t.dyn_cast<RankedTensorType>();
-    return tensorType && tensorType.getEncoding();
-  };
-  auto valueHasEncoding = [=](Value v) -> bool {
-    return typeHasEncoding(v.getType());
-  };
   // Mark any operation that has operands/results with encoding as
   // illegal.
-  auto hasOperandOrResultsWithEncoding = [=](Operation *op) {
-    return llvm::any_of(op->getOperands(), valueHasEncoding) ||
-           llvm::any_of(op->getResultTypes(), typeHasEncoding);
-  };
-  markUnknownOpDynamicallyLegal(
-      [&](Operation *op) { return !hasOperandOrResultsWithEncoding(op); });
+  markUnknownOpDynamicallyLegal([=](Operation *op) {
```

Comment:
Is this correct. Where is the `typeHasEncoding` method defined?

Reply:
oh lol, sorry. so there's actually a fair use for at least the `typeHasEncoding` lambda here. this tips the scale back to some lambda + any_of usage then, how about this.

Reply:
Actually @MaheshRavishankar i pushed another commit replacing two of the `[=]` by `[]` (capture no context at all).